### PR TITLE
add styled-components example

### LINKED ISSUE #39: styled-components integration?
It would be nice to have a styled-component integration...
Any ideas?

### PR DESCRIPTION
**Pull request by mathieudutour**
I played around adding `react-primitives` to `styled-components` which means that you can use it in react-sketchapp as well!

fix #39

### CODE REVIEW

**Comment by lloydwheeler** [comment]
Thanks for this! I've done a quick test on this today - hopefully OK me responding even though I'm not a maintainer. 🤔

Using the `styled-components` example when rendering to sketch I get a RedBox error `ReferenceError: Can't find variable: StyleSheet`. I also tried using the browser as a target with a simple example:

```
import React from 'react';
import { View, Text } from 'react-primitives';
import styled from 'styled-components/primitives';

const StyledTitle = styled.Text`
  color: red;
`;

export default () => (
  <StyledTitle>Test</StyledTitle>
);
```

and get `Uncaught TypeError: StyleSheet.create is not a function`.

**Comment by mathieudutour** [comment]
oh I think I didn't push the latest version on the `styled-components` fork.

Could you try again now? (I'm really sorry, I can't seem to make any example work right now, probably something wrong with my config)

**Comment by lloydwheeler** [comment]
No problem! Grabbed the latest update but seeing a few issues still when rendering to Sketch. 

1. `Error: Failed to parse declaration "margin: 4,"`
2. `Error: Failed to parse declaration "padding: 8,"`
3. `Error: Failed to parse declaration "fontWeight: 'bold',"`

If I remove all references to those declarations I get 

```
<MOMethod: 0x608001428bc0 : target=0x60800225bb10-[MOUndefined doubleValue]: unrecognized selector sent to instance 0x60000000bd10, selector=name>: undefined
```

Running the basic web example from the previous comment works fine though!

**Comment by mathieudutour** [comment]
I'm an idiot, I didn't actually write css in the template strings ;)

File: examples/styled-components/src/my-command.js [hunk 1]
```diff
@@ -15,20 +15,20 @@ const textColor = (hex) => {
...
+  margin: 4px;
+  background-color: ${props => props.hex};
+  padding: 8px;
 `;
 
 const SwatchName = styled.Text`
-  color: ${props => textColor(props.hex)},
-  fontWeight: 'bold',
+  color: ${props => textColor(props.hex)};
+  font-weight: 'bold';
```
Comment (lloydwheeler) [inline examples/styled-components/src/my-command.js]: Need to remove the `''` around the property value here.

**Comment by lloydwheeler** [comment]
With that last fix on the `font-weight` property it all seems to render to Sketch fine. Looking forward to using this! 🙏

**Comment by mathieudutour** [comment]
It does? Awesome! Thank you so much for your help. Programming blindly without testing is pretty hard :)

**Comment by jongold** [comment]
this is awesome, really excited to see it.

I can't talk from the styled-components side of the fence but consider this mergeable if / when SC publish it :)

Do we want a different example to make it distinct? I made this - thoughts?
image

File: examples/styled-components/package.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    "skpm": "^0.9.0"
+  },
+  "dependencies": {
+    "chroma-js": "^1.2.2",
+    "prop-types": "^15.5.8",
+    "react": "^15.4.2",
+    "react-primitives": "^0.4.2",
+    "react-sketchapp": "^0.11.0",
+    "react-test-renderer": "^15.4.2",
+    "styled-components": "git+https://github.com/mathieudutour/styled-components.git#built"
```
Comment (jongold) [inline examples/styled-components/package.json]: Hold off merging until this gets merged to styled-components

**Comment by jongold** [comment]
Oops - accidentally pushed my `Max's Sweaters` design in my last commit - feel free to revert it, up to you

**Comment by mxstbr** [comment]
:shipit:

**Pull request merged**